### ISSUE TITLE
[Bug]: ollama streaming example fails

### ISSUE THREAD

**Issue by altruios** [BerriAI/litellm#1191]
### What happened?

I try running the demo code.
```
from litellm import completion

response = completion(
    model="ollama/llama2", 
    messages=[{ "content": "respond in 20 words. who are you?","role": "user"}], 
    api_base="http://localhost:11434",
    stream=True
)
for chunk in response:
    print(chunk)


```
Digging further I found the culprit to be an object/dict shaped like:

{'role': 'assistant', 'content': '/n'}

...looks like it wants this shape:

{str:response,bool:done}

...instead.

### Relevant log output

```shell
Give Feedback / Get Help: https://github.com/BerriAI/litellm/issues/new
LiteLLM.Info: If you need to debug this error, use `litellm.set_verbose=True'.

Traceback (most recent call last):
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5828, in chunk_creator
    response_obj = self.handle_ollama_stream(chunk)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5651, in handle_ollama_stream
    raise e
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5632, in handle_ollama_stream
    json_chunk = json.loads(chunk)
                 ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 339, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not dict

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "{dir}/main.py", line 5, in <module>
    for chunk in say_hello("cat"):
                 ^^^^^^^^^^^^^^^^
  File "{dir}/lib/python3.11/site-packages/magentic/prompt_function.py", line 79, in __call__
    message = self.model.complete(
              ^^^^^^^^^^^^^^^^^^^^
  File "{dir}/lib/python3.11/site-packages/magentic/chat_model/litellm_chat_model.py", line 215, in complete
    first_chunk = next(response)
                  ^^^^^^^^^^^^^^
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5946, in __next__
    raise e
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5932, in __next__
    response = self.chunk_creator(chunk=chunk)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5919, in chunk_creator
    raise exception_type(model=self.model, custom_llm_provider=self.custom_llm_provider, original_exception=e)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5192, in exception_type
    raise e
  File "{dir}/lib/python3.11/site-packages/litellm/utils.py", line 5169, in exception_type
    raise APIConnectionError( 
    ^^^^^^^^^^^^^^^^^^^^^^^^^
litellm.exceptions.APIConnectionError: the JSON object must be str, bytes or bytearray, not dict
```

**Comment by qrkourier** [comment]
I believe I triggered this problem with this litellm-to-ollama proxy config.

```yaml
model_list:
- litellm_params:
    api_base: http://ollama:11434
    model: ollama/hub/linux-terminal:latest
    stream: true
  model_name: hub/linux-terminal:latest
```

The error did not recur with `stream: false`.

```
 final response: <async_generator object _async_streaming at 0x7fc6bb2f1ee0>
 INFO:     172.28.0.4:38916 - "POST /chat/completions HTTP/1.1" 500 Internal Server Error
 ERROR:    Exception in ASGI application
 Traceback (most recent call last):
   File "/usr/local/lib/python3.9/site-packages/fastapi/encoders.py", line 230, in jsonable_encoder
     data = dict(obj)
 TypeError: 'async_generator' object is not iterable

 During handling of the above exception, another exception occurred:

 Traceback (most recent call last):
   File "/usr/local/lib/python3.9/site-packages/fastapi/encoders.py", line 235, in jsonable_encoder
     data = vars(obj)
 TypeError: vars() argument must have __dict__ attribute

 The above exception was the direct cause of the following exception:

 Traceback (most recent call last):
   File "/usr/local/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
     result = await app(  # type: ignore[func-returns-value]
   File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
     return await self.app(scope, receive, send)
   File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 292, in __call__
     await super().__call__(scope, receive, send)
   File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 122, in __call__
     await self.middleware_stack(scope, receive, send)
   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
     raise exc
   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
     await self.app(scope, receive, _send)
   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/cors.py", line 83, in __call__
     await self.app(scope, receive, send)
   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
     raise exc
   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
     await self.app(scope, receive, sender)
   File "/usr/local/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
     raise e
   File "/usr/local/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
     await self.app(scope, receive, send)
   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 718, in __call__
     await route.handle(scope, receive, send)
   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
     await self.app(scope, receive, send)
   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
     response = await func(request)
   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 291, in app
     content = await serialize_response(
   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 179, in serialize_response
     return jsonable_encoder(response_content)
   File "/usr/local/lib/python3.9/site-packages/fastapi/encoders.py", line 238, in jsonable_encoder
     raise ValueError(errors) from e
 ValueError: [TypeError("'async_generator' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
Internal Server Error%

```

**Comment by ghost** [comment]
Hey @altruios can you confirm if this error persists in litellm v`1.15.3`?

**Comment by altruios** [comment]
appears to work! current output is:

```
ModelResponse(id='chatcmpl-0e718554-8296-404d-a4fa-7af80053b597', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content='I', role='assistant'))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-3dad45cf-5589-4084-adde-1afa99e1fe24', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content="'", role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-0b11eefa-e050-4db2-a5dd-372cc2952376', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content='m', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-f8f07f95-7381-4718-bf98-a921e61e7357', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=' just', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-10fe8131-5b31-4b3f-83bf-0f4dd5530fe7', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=' an', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-d2bb2990-69f3-4db5-a55a-b39926540dd3', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=' A', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-f1922e22-a957-478f-8aef-e8e690f225bd', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content='I', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-500c4e93-a06c-444f-8628-a061e167c2d9', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=' assistant', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-0efd6cb4-73cc-4d7c-8493-d6ea1075600c', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=',', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-dc887809-dcc0-42e9-8015-fdd7668ea586', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=' here', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-b0b81372-0a94-4227-b7c2-11555ebd04f2', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=' to', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-03a89c57-510b-4def-8250-4142a11b450d', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content=' help', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-6f8413d2-a389-4822-9da4-8ed70169d093', choices=[StreamingChoices(finish_reason=None, index=0, delta=Delta(content='!', role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())
ModelResponse(id='chatcmpl-abd46417-741f-49f7-b29c-3512447a642c', choices=[StreamingChoices(finish_reason='stop', index=0, delta=Delta(content=None, role=None))], created=1703207003, model='llama2', object='chat.completion.chunk', system_fingerprint=None, usage=Usage())

```

**Comment by ishaan-jaff** [comment]
fixed in 1.15.3
